Library/Homebrew/cmd/update-bash.sh

```diff
@@ -263,15 +282,16 @@ EOS
     odie "$HOMEBREW_REPOSITORY must be writable!"
   fi
 
-  if [[ -z "$(which_git)" ]]
+  HOMEBREW_GIT="$(which_git)"
+  if [[ -z "$HOMEBREW_GIT" ]]
```

Comment:
Can you do the assignment inside the `[[]]` somehow? I didn't think so but figured I'd ask CC @UniqMartin